avcastoldi/fuba
# [Agile/Devops] Blank Rails Application, ready to CI/CD

This branch contains the following functionality:
* #1 - As an Entrepreneur I want to quickly create applications that are easy to maintain
* #2 - As an Entrepreneur I want to code using the Red-Green-Refactor pattern
* #3 - As an Entrepreneur I want to follow the language best practices

## Documentation
* Does this Pull Request adds a Blank Rails 5 application.
* It uses CodeStar as CI/CD tool. The solution is deployed into a Elastic Beanstalk Cluster. I could deploy it to Heroku, but it's old school. I choose CodeStar to show that I'm a Cloud Rock Star 🎉 (just kidding)
* Classical static analysis tools to reinforce good coding practices were also introduced by this PR: `guard`, `rubocop`, `reek` and `simplecov`.

## How to test
1) Ensure all package dependencies were met:
```
bundle install
```
2) Run the test suite
```
rails test
```
3) Launch the web server
```
rails server
```
4) Visit http://localhost:3000. You should the project land page
## Screenshots / Recordings
* Test results
![image](https://user-images.githubusercontent.com/31883/39741829-2d6de3ea-5250-11e8-8621-774c0c4cc873.png)

* Land page
![image](https://user-images.githubusercontent.com/31883/39741794-12e24cd2-5250-11e8-8cfb-0f9f5a98506b.png)
